common/python/ax/cluster_management/argo_cluster_manager.py

```diff
@@ -102,6 +115,54 @@ def uninstall(self, args):
         self._ensure_customer_id(uninstall_config.cloud_profile)
         ClusterUninstaller(uninstall_config).start()
 
+    def install_platform_only(self, args):
+        os.environ["AX_CUSTOMER_ID"] = "user-customer-id"
+        os.environ["ARGO_LOG_BUCKET_NAME"] = args.cluster_bucket
+        os.environ["ARGO_DATA_BUCKET_NAME"] = args.cluster_bucket
+        os.environ["ARGO_KUBE_CONFIG_PATH"] = args.kubeconfig
+
+        logger.info("Using customer id: %s", os.environ["AX_CUSTOMER_ID"])
+
+        args.silent = True
+        # 1. Create the platform install config first.
+        platform_install_config = PlatformOnlyInstallConfig(cfg=args)
+
+        config.load_kube_config(config_file=args.kubeconfig)
+        v1 = client.CoreV1Api()
+        ret = v1.list_node(watch=False)
+        assert len(ret.items) > 0, "There should at least be 1 node in the Kubernetes cluster"
+        instance = bunchify(ret.items[0])
+        # TODO(shri): Add labels to the nodes here?
+
+        # 2. Populate the config with region and placement from the K8S cluster
+        platform_install_config.region = instance.metadata.labels["failure-domain.beta.kubernetes.io/region"]
```

Comment:
Initially, users would probably have to run on single region, single AZ clusters anyway. Otherwise, not sure how pods, volumes, ec2 instances, etc. would work across zones; especially, since this hasn't been tested. I understand that K8S says that it does the right thing and scheduled pods according to the AZ in which it's PV/PVC is, but would be good test it.

I've removed populating the region and AZ from the K8S labels.

Reply:
pls make sure that cluster can live without specifying a region, as I remember some components have `AX_REGION` env.